youtube_dlc/downloader/niconico.py

```diff
@@ -0,0 +1,26 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .http import HttpFD
+from .external import FFmpegFD
+from ..extractor.niconico import NiconicoIE
+
```

Comment:
It might be helpful to create a base class `NiconicoDmcFD` and derive these 2 classes from it. Like this:

    class NiconicoDmcHttpFD(NiconicoDmcFD, HttpFD):

It's just a suggestion. You don't need to do it if you think having separate classes is easier

Reply:
Wow!

Reply:
> Wow!

?

Reply:
Oh, sorry. I have not-good image about multiple inheritence. But, python's it is very elegant, I surprised.